File: check_avs.js
function canonize(avs_no) {
	return avs_no.replace(/\./g, "")
}

function is_valid_avs_number (avs_no) {
  avs_no = canonize(avs_no);
	if (avs_no.length != 13) return false;
  
  var total = 0;
  
  for(var i = 0 ; i < 12 ; i+=2) {
      total += parseInt(avs_no[i]);
  }
  
  for (var i = 1 ; i < 12 ; i+=2) {
  		total += parseInt(avs_no[i]) * 3;
  }
  
  var expected_key = 0;
  var actual_key = parseInt(avs_no[12]);
  if (total % 10 != 0) {
      var round_ten_up = Math.floor(total/10) * 10 + 10;
      expected_key = round_ten_up - total;
  }
  
  return actual_key == expected_key;
}

File: CheckAvs.java
public class CheckAvs {

    private static String canonize(String avsNo) {
        return avsNo.replace(".", "");
    }

    public static boolean isValidAvsNumber(String avsNo) {
        avsNo = canonize(avsNo);

        if (avsNo.length() != 13) return false;

        int total = 0;

        for(int i = 0 ; i < 12 ; i+=2) {
            total += digitAt(avsNo, i);
        }

        for(int i = 1 ; i < 12 ; i+=2) {
            total += 3 * digitAt(avsNo, i);
        }

        int actualKey = digitAt(avsNo, 12);
        int expectedKey = 0;

        if (total % 10 != 0) {
            int roundTenUp = (total / 10) * 10 + 10;
            expectedKey = roundTenUp - total;
        }

        return expectedKey == actualKey;
    }

    private static int digitAt(String str, int i) {
        return Integer.parseInt(String.valueOf(str.charAt(i)));
    }

    public static void main(String[] args) {
        String avsNo = args[0];
        System.out.println(isValidAvsNumber(avsNo) ? "Valid" : "Invalid");
    }
}
